service/ExchangeService.js

```diff
@@ -7,20 +7,34 @@ var btcPrice   = null;
 var krwUsd     = null;
 var log = new Logger(AppConfig.logLevel);
 
-
+var createErrorExtension = function(message, exchange, errortype){
+	var errorObject = new Error();
+		errorObject.message = message;
+		errorObject.exchange = exchange;
+		errorObject.errorType = errortype;
+	return errorObject;
+}
 var fetchExchangeData = function(exchange, url, callback){
 	request.get(url, function (err, response, body) {
-		if ( !err && response.statusCode == 200 ){
-			try {
-				callback(null, JSON.parse(body));
-			}catch (e) {
-				return callback('ERROR parsing response from ' + exchange + '. Details: ' + e + '\nResponse::' + body);
-			}
-		}else if ( err ){
-			return callback('ERROR: fetching data from ' + exchange + ' (' + url + ') Details: ' + err + '. BODY=' + body, null);
-		}else{
-			return callback('ERROR: unexpected response code (' + response.statusCode + ') from url=[' + url + '] Details: ' + body);
+		if ( err ){
+			var errorFetching = 'ERROR: fetching data from ';
+			log.debug(errorFetching + exchange + ' (' + url + ') Details: ' + err + '. BODY=' + body, null);
+			return callback(createErrorExtension(errorFetching, exchange, err), null);
```

Comment:
It's good practice to have a constant type of return value. In this case, we return callback's result and don't control it.